### DISCUSSION TITLE
Expected 2D array, got scalar array instead: array=X_test. Reshape your data either using array.reshape(-1, 1) if your data has a single feature or array.reshape(1, -1) if it contains a single sample.

### DISCUSSION THREAD

**Discussion by talhash02** [AammarTufail/python_ka_chilla2023#84] [Q&A]
# libraries and model import

import numpy as np
import pandas as pd
import seaborn as sns
df= sns.load_dataset('planets')
X = df[['number','orbital_period','orbital_period','distance','year']]
y=df['method']

from sklearn.model_selection import train_test_split, cross_val_score

from sklearn.metrics import accuracy_score,f1_score,precision_score, recall_score
from sklearn.linear_model import LogisticRegression
from sklearn.neighbors import KNeighborsClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.svm import SVC
from sklearn.tree import DecisionTreeClassifier 

# Now we will split the datasets for training & testing using Train_test_split library

X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.2, random_state=42)
# define the models to be evaluated
models = [LogisticRegression(), KNeighborsClassifier(), RandomForestClassifier(), SVC(), DecisionTreeClassifier()]
model_name = ["Logistic Regression", "KNN", "Random Forest", "SVM","Decision Clasifier"] 

model_scores = []
 
# perform k-fold cross-validation for each model
for  models,model_name in zip(models, model_name):
    # Train the model
    models.fit(X_train, y_train)
    # predict the model by X_test remaning 20% values.
    y_pred = models.predict('X_test')
    
    acccuracy = accuracy_score(y_test, y_pred)
   
    model_scores.append([model_name,acccuracy])

    sorted_models = sorted(model_scores, key=lambda x:x[1], reverse= True)

for model in sorted_models:
    print("Accuracy Score: ",f'{model[0]} : {model[1]:.2f}')

Please help me to identify error.

**Comment by usamabajwa86** [comment]
use get_dummies command to reshape your data from object to numeric
